backend/generator.js

```diff
@@ -9,41 +9,114 @@ exports.executeScript = function (socket, formData) {
     socket.emit('err-logs', "Failed to generated documentation due to an error in input fields");
     return false;
   }
-  var email = formData.email;
-  var gitUrl = formData.gitUrl;
-  var docTheme = formData.docTheme;
-  var uniqueId = uuidV4();
-  var webUI = "true"
-
-  var donePercent = 0;
-
-  const args = [
-    "-g", gitUrl,
-    "-t", docTheme,
-    "-m", email,
-    "-u", uniqueId,
-    "-w", webUI
-  ];
-
-  var process = spawn("./generate.sh", args);
-
-  process.stdout.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('logs', {donePercent: (donePercent = donePercent + 4), data: data.toString()});
-  });
-
-  process.stderr.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('err-logs', data.toString());
-  });
-
-  process.on('exit', function (code) {
-    console.log('child process exited with code ' + code);
-    if (code === 0) {
-      socket.emit('success', {email: email, uniqueId: uniqueId, gitUrl: gitUrl});
-    } else {
-      socket.emit('failure', {errorCode: code});
-    }
-  });
-  return true;
+  // logic for one Repository
```

Comment:
all the build process are happening in the bash script only, I just added a condition for a single repository or multiple repositories. We don't know how many features will be implemented in future. when we keep on adding thing to a single file, we making things much complex and also making hard to read as well as maintain. still we don't know building documentation for multiple repositories is our end goal. even i'm reusing old scripts like toc tree generation. we'll maintain separate script for WEBUI and travis. as discussed @imujjwal96 will work on Travis implementation